### PR TITLE
test: setup for ts tests

### PR DESCRIPTION
**Pull request by ErinHales**
- migrate HardhatRuntimeEnvironment
 - add deploy for test contracts
 - test Common

### CODE REVIEW

**Review by prestwich — CHANGES_REQUESTED** [review]
okay, some good work here. I'd like to see us split up the tests and the dev deploys to a new typescript package. `typescript/optics-deploy` and `typescript/optics-test`

**Comment by ErinHales** [comment]
> okay, some good work here. I'd like to see us split up the tests and the dev deploys to a new typescript package. `typescript/optics-deploy` and `typescript/optics-test`

Yep, that definitely makes sense.  I refactored the typescript directory a bit.  I didn't rename the `test` directory to `optics-test` though, because hardhat is stubborn

**Comment by prestwich** [comment]
do you think it makes sense to have these be entirely separate packages? each with a package.json?

Then only the optics-test package would need hardhat

**Comment by ErinHales** [comment]
> do you think it makes sense to have these be entirely separate packages? each with a package.json?
> 
> Then only the optics-test package would need hardhat

Possibly? Do you think we'd want to make optics-deploy into an npm package?

File: typescript/test/lib/index.ts [hunk 1]
```diff
@@ -0,0 +1,283 @@
...
+import * as deployHelpers from '../../optics-deploy/deployOptics';
+import { getHexStringByteLength } from './utils';
+<<<<<<< Updated upstream
+import { TestHome__factory, TestReplica__factory } from '../../typechain/optics-core';
+=======
+import {
+  TestHome__factory,
+  TestReplica__factory,
+} from '../../typechain/optics-core';
+>>>>>>> Stashed changes
```
Comment (YoduYodu) [inline typescript/test/lib/index.ts:16]: ☝️

**Comment by ErinHales** [comment]
TODO:Install packages in `typescript/typechain`

**Comment by ErinHales** [comment]
Closing in favor of #400

**Pull request closed without merging**